src/nvim/api/vim.c

```diff
@@ -695,6 +696,73 @@ Dictionary nvim_get_color_map(void)
   return colors;
 }
 
+// Helper function for nvim_get_keymap and nvim_buf_get_keymap
+// @param mode The abbreviation for the mode
+// @param buffer_id
+//      -1: Global mapping
+//      0: Current buffer
+//      else: The desired buffer ID
+//
+// @returns An array of maparg() like dictionaries describing mappings
+static Array get_keymap_helper(String mode, Integer buffer_id)
+{
+  Array mappings = ARRAY_DICT_INIT;
+  dict_T *const dict = dict_alloc();
+
+  // Convert the string mode to the integer mode
+  // that is stored within each mapblock
+  int int_mode = get_map_mode((char_u **)(&(mode.data)), 0);
+
+  for (int i = 0; i < MAX_MAPHASH; i++) {
+    for (mapblock_T *current_maphash = get_maphash(i, (long)buffer_id);
+        current_maphash;
```

Comment:
Too dedented: should be aligned with `current_maphash`. Would also suggest using `const mapblock_T *`.